src/background.ts

```diff
@@ -6,15 +6,34 @@
     const calendar_name = "AutoCalendar Demo";
     const autocalendar = new AutoCalendar();
 
-    chrome.runtime.onMessage.addListener(
-        async (request, sender, sendResponse) => {
-            if (request.action === "sync_canvas") {
+    let prev_execution = null;
+    let isRunning = false;
+
+    chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+        if (request.action === "oauth") {
+            autocalendar.calendar.session.oauth_token();
+        } else if (request.action === "sync_canvas") {
+            let time_diff =
+                prev_execution === null
+                    ? Number.MAX_VALUE
+                    : new Date().getTime() - prev_execution.getTime();
+            if (!isRunning && time_diff > 62000) {
```

Comment:
This should be 60 seconds, leaving a 2 second buffer seems pointless to me

Reply:
Solved!